tests/subsys/dfu/dfu_target_stream/src/main.c

```diff
@@ -285,9 +289,9 @@ static void test_dfu_target_stream_save_progress(void)
 
 void test_main(void)
 {
-	fdev = device_get_binding(FLASH_NAME);
-
 #ifdef CONFIG_DFU_TARGET_STREAM_SAVE_PROGRESS
+	__ASSERT_NO_MSG(device_is_ready(fdev));
```

Comment:
Why is this in the ifdef? `test_dfu_target_stream_null_checks` uses `fdev` and is run outside the ifdef.

Reply:
because `fdev` is used just below in this case, outside the tests.

Reply:
this is `test_main`, so it's the first code to be executed, before any test. if you check L298, `fdev` is used, so we need to assert for it to be ready.